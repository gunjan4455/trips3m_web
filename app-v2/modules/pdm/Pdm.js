import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { getSelectedPdms } from 'helpers/pdmHelpers';
import { PdmContainer } from './G';

class Pdm extends React.Component {

  static propTypes = {
    setMessage: PropTypes.func.isRequired,
    pdmsData: PropTypes.object.isRequired,
    selectedPdm: PropTypes.string.isRequired,
  };

  static defaultProps = {
    setMessage: () => {
    }
  };

  constructor(props) {
    super(props);
    this.onPdmSelection = this.onPdmSelection.bind(this);
  }

  onPdmSelection(pdm) {
    this.props.setMessage(pdm.message_text);
  }

  render() {
    const { pdmsData, selectedPdm } = this.props;
    const pdmList = getSelectedPdms(pdmsData, selectedPdm);

    if (!pdmList || !pdmList.length) {
      return null;
    }

    return (
      <PdmContainer className="clearfix relative">
        <ul>
          {
            pdmList.map((item, index) =>
              <li key={index} >
                <a className={`at_${index}`} onClick={() => this.onPdmSelection(item)}>
                  {item.pdm_text}
                </a>
              </li>
            )
          }
        </ul>
      </PdmContainer>
    );
  }
}

export default Pdm;

