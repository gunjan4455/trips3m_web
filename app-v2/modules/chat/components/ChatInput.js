import React from 'react';
import PropTypes from 'prop-types';

import {
  AttachmentIcon, KeyboardActive, KeyboardInactive, SendIcon, SendIconGrey
} from 'helpers/Icon/Icon';
import Pdm from 'modules/pdm';
import { GKeyBoard, GPdmContainer } from '../G';

class ChatInput extends React.Component {
  static propTypes = {
    pdmsInformation: PropTypes.object.isRequired,
    sendMessage: PropTypes.func.isRequired,
    showHidePdmView: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.handleKeyUp = this.handleKeyUp.bind(this);
    this.messageOnChange = this.messageOnChange.bind(this);
    this.setPdmMessage = this.setPdmMessage.bind(this);
    this.getVariationSendButton = this.getVariationSendButton.bind(this);
    this.focusOnInput = this.focusOnInput.bind(this);
    this.state = {
      messageText: '',
    };
  }

  componentDidMount() {
    window.addEventListener('QDP_CHAT_EX_MSG', (event) => {
      console.log('event: ', event);
      this.setState({
        messageText: event.detail.message
      });
      event.detail.focusOnInput && this.focusOnInput();
    });
  }

  focusOnInput() {
    if (this.chatInput) {
      this.chatInput.focus();
    }
  }

  showHidePdmView() {
    this.props.showHidePdmView(!this.props.pdmsInformation.showPdmView);
  }

  handleClick() {
    const message = this.state.messageText;

    if (message) {
      this.props.sendMessage(message);
    }

    this.setState({
      messageText: ''
    });

    this.focusOnInput();
  }

  setPdmMessage(pdmMessage) {
    const message = this.state.messageText;
    this.setState({
      messageText: pdmMessage
    });

    this.showHidePdmView();
    this.focusOnInput();
  }

  handleKeyUp(e) {
    if (e.nativeEvent.keyCode === 13) {
      e.preventDefault();
      this.handleClick();
    }
  }

  messageOnChange(e) {
    if (e.target.value !== '\n') {
      this.setState({ messageText: e.target.value, });
    }
  }

  getVariationSendButton() {
    return (
      <button className="attachment-button at_chatsendbtn absolute b10 r15" onClick={this.handleClick}>Send
      </button>);
  }

  render() {
    const { pdmsInformation } = this.props;

    return (
      <div
        className="row row- relative type-chat-con mb0 at_chatmessage">
        {
          pdmsInformation.showPdmView ?
            <GPdmContainer>
              <Pdm setMessage={this.setPdmMessage} />
            </GPdmContainer> : null
        }

        <button className="hide absolute attachment-button"><AttachmentIcon /></button>
        <div className="clearfix">
          {
            pdmsInformation.pdms ?
              <GKeyBoard className="l0 absolute-center-v" onClick={() => this.showHidePdmView()}>
                <span className="pl4 pl4 pt8 pb8 block">
                  {pdmsInformation.showPdmView ? <KeyboardActive /> : <KeyboardInactive />}
                  </span>
              </GKeyBoard>
              : null
          }
          <textarea
            onKeyPress={this.handleKeyUp}
            onKeyUp={this.handleKeyUp}
            className="at_chatinput"
            ref={(input) => {
              this.chatInput = input;
            }}
            onChange={this.messageOnChange}
            value={this.state.messageText}
            placeholder="Type a messageâ€¦" />
          {this.getVariationSendButton()}
        </div>
      </div>
    );
  }
}

export default ChatInput;

