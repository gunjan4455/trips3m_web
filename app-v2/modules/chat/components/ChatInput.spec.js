import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ChatInput from './ChatInput';
import Pdm from 'modules/pdm';

configure({ adapter: new Adapter() });

const sendMessageMock = jest.fn();

const DEFAULT_PROPS = {
  pdmsInformation: {
    showPdmView: true,
    pdms: {}
  },
  sendMessage: sendMessageMock,
  showHidePdmView: jest.fn,
};

const shallowWrappperChatInput = (props) => {
  return shallow(<ChatInput {...props} />);
};

describe('ChatInput: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
    jest.clearAllMocks();
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperChatInput(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render pdm component when showPdmView is false ', () => {
    props.pdmsInformation.showPdmView = false;
    const wrapper = shallowWrappperChatInput(props);
    expect(wrapper.find(Pdm)).toHaveLength(0);
  });

  it('should call messageOnChange on type', () => {
    props.pdmsInformation.showPdmView = false;

    const wrapper = shallowWrappperChatInput(props);

    const spy = jest.spyOn(wrapper.instance(), "messageOnChange");
    wrapper.instance().forceUpdate();

    const e = { target: { value: 'My new value' } };
    wrapper.find('.at_chatinput').simulate('change', e);

    expect(spy).toBeCalledWith(e);
  });


  it('should call handleKeyUp when enter key is pressed', () => {
    const wrapper = shallowWrappperChatInput(props);

    const spy = jest.spyOn(wrapper.instance(), "handleKeyUp");
    wrapper.instance().forceUpdate();

    wrapper.find('.at_chatinput').simulate('keyUp', { nativeEvent: { keyCode: 13 }, preventDefault: jest.fn });

    expect(spy).toHaveBeenCalled();
  });

  it('should call sendMessage when enter key is pressed with message', () => {
    const wrapper = shallowWrappperChatInput(props);

    const e = { target: { value: 'My new value' } };
    wrapper.find('.at_chatinput').simulate('change', e);
    wrapper.find('.at_chatsendbtn').simulate('click');

    expect(sendMessageMock).toHaveBeenCalledWith('My new value');
  });

});
