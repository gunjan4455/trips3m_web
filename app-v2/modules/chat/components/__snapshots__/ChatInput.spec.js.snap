// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatInput
    pdmsInformation={
      Object {
        "pdms": Object {},
        "showPdmView": true,
      }
    }
    sendMessage={[MockFunction]}
    showHidePdmView={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(render)>
          <withRouter(Connect(Pdm))
            setMessage={[Function]}
          />
        </ForwardRef(render)>,
        <button
          className="hide absolute attachment-button"
        >
          <IconSvg />
        </button>,
        <div
          className="clearfix"
        >
          <ForwardRef(render)
            className="l0 absolute-center-v"
            onClick={[Function]}
          >
            <span
              className="pl4 pl4 pt8 pb8 block"
            >
              <IconSvg />
            </span>
          </ForwardRef(render)>
          <textarea
            className="at_chatinput"
            onChange={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="Type a message…"
            value=""
          />
          <button
            className="attachment-button at_chatsendbtn absolute b10 r15"
            onClick={[Function]}
          >
            Send
          </button>
        </div>,
      ],
      "className": "row row- relative type-chat-con mb0 at_chatmessage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <withRouter(Connect(Pdm))
            setMessage={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setMessage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "bottom": "100%",
              "left": "0",
              "position": "absolute",
              "right": "0",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "hide absolute attachment-button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef(render)
              className="l0 absolute-center-v"
              onClick={[Function]}
            >
              <span
                className="pl4 pl4 pt8 pb8 block"
              >
                <IconSvg />
              </span>
            </ForwardRef(render)>,
            <textarea
              className="at_chatinput"
              onChange={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="Type a message…"
              value=""
            />,
            <button
              className="attachment-button at_chatsendbtn absolute b10 r15"
              onClick={[Function]}
            >
              Send
            </button>,
          ],
          "className": "clearfix",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="pl4 pl4 pt8 pb8 block"
              >
                <IconSvg />
              </span>,
              "className": "l0 absolute-center-v",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg />,
                "className": "pl4 pl4 pt8 pb8 block",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "button",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                Object {
                  "& > span": Object {
                    "& svg": Object {
                      "height": "13px !important",
                      "width": "26px !important",
                    },
                    "height": "30px",
                    "width": "36px",
                  },
                  "border": "0 none",
                  "height": "30px",
                  "width": "36px",
                  "zIndex": "2",
                },
              ],
              "defaultProps": undefined,
              "displayName": "Styled(button)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "at_chatinput",
              "onChange": [Function],
              "onKeyPress": [Function],
              "onKeyUp": [Function],
              "placeholder": "Type a message…",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send",
              "className": "attachment-button at_chatsendbtn absolute b10 r15",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Send",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(render)>
            <withRouter(Connect(Pdm))
              setMessage={[Function]}
            />
          </ForwardRef(render)>,
          <button
            className="hide absolute attachment-button"
          >
            <IconSvg />
          </button>,
          <div
            className="clearfix"
          >
            <ForwardRef(render)
              className="l0 absolute-center-v"
              onClick={[Function]}
            >
              <span
                className="pl4 pl4 pt8 pb8 block"
              >
                <IconSvg />
              </span>
            </ForwardRef(render)>
            <textarea
              className="at_chatinput"
              onChange={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="Type a message…"
              value=""
            />
            <button
              className="attachment-button at_chatsendbtn absolute b10 r15"
              onClick={[Function]}
            >
              Send
            </button>
          </div>,
        ],
        "className": "row row- relative type-chat-con mb0 at_chatmessage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withRouter(Connect(Pdm))
              setMessage={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "setMessage": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "bottom": "100%",
                "left": "0",
                "position": "absolute",
                "right": "0",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "hide absolute attachment-button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef(render)
                className="l0 absolute-center-v"
                onClick={[Function]}
              >
                <span
                  className="pl4 pl4 pt8 pb8 block"
                >
                  <IconSvg />
                </span>
              </ForwardRef(render)>,
              <textarea
                className="at_chatinput"
                onChange={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                placeholder="Type a message…"
                value=""
              />,
              <button
                className="attachment-button at_chatsendbtn absolute b10 r15"
                onClick={[Function]}
              >
                Send
              </button>,
            ],
            "className": "clearfix",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <span
                  className="pl4 pl4 pt8 pb8 block"
                >
                  <IconSvg />
                </span>,
                "className": "l0 absolute-center-v",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg />,
                  "className": "pl4 pl4 pt8 pb8 block",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "button",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "& > span": Object {
                      "& svg": Object {
                        "height": "13px !important",
                        "width": "26px !important",
                      },
                      "height": "30px",
                      "width": "36px",
                    },
                    "border": "0 none",
                    "height": "30px",
                    "width": "36px",
                    "zIndex": "2",
                  },
                ],
                "defaultProps": undefined,
                "displayName": "Styled(button)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "at_chatinput",
                "onChange": [Function],
                "onKeyPress": [Function],
                "onKeyUp": [Function],
                "placeholder": "Type a message…",
                "value": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send",
                "className": "attachment-button at_chatsendbtn absolute b10 r15",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Send",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
